sysctl.d/50-default.conf: remove *, .all source route settings

t-lo opened the pull request:
# Remove wildcard, "all" source route settings from sysctl defaults

The rules were added in systemd-245 and break cluster networking, e.g. cilium. Please see https://github.com/flatcar-linux/Flatcar/issues/181 for details.

Signed-off-by: Thilo Fromm <thilo@kinvolk.io>

### Also cherry-pick to `v246-flatcar` which has the same issue.

t-lo committed:
sysctl.d/50-default.conf: remove *, .all source route settings

The rules were added in systemd-245 and break cluster
networking, e.g. cilium. Please see
https://github.com/flatcar-linux/Flatcar/issues/181
for details.

Signed-off-by: Thilo Fromm <thilo@kinvolk.io>

margamanterola commented on sysctl.d/50-default.conf:
```diff
@@ -23,18 +23,27 @@ kernel.core_uses_pid = 1
 net.ipv4.conf.default.rp_filter = 2
-net.ipv4.conf.*.rp_filter = 2
--net.ipv4.conf.all.rp_filter
+# the below deviates from upstream systemd-245 (and later) since the default
```
Can you add a noun between "the" and "below"?
e.g. "settings", "rules", "configurations", etc

Same in the other two comments.

t-lo replied:
But... below *is* the [noun](https://www.merriam-webster.com/dictionary/below). I get your point wrt. being more specific though, will update.

margamanterola reviewed (commented):
Is commenting the rules really enough? Shouldn't we set them to the opposite?

t-lo committed:
sysctl.d/50-default: better comments, re-activate promote_secondaries

This change updates comments as well as re-activates the
promote_secondaries wildcard since networkd's DHCP relies on this
(see https://github.com/systemd/systemd/issues/7163)

Signed-off-by: Thilo Fromm <thilo@kinvolk.io>

t-lo commented:
Agree on `accept_source_route` and will update the PR.
Also agree with revisiting this if it does not fix the cilium issue - however since adding an override rule which only sets `rp_filter=0` seems to [work around the cilium the issue](https://github.com/flatcar-linux/Flatcar/issues/181) I'm optimistic that we don't need to.

t-lo committed:
sysctl.d/50-default.conf: re-activate default accept_source_route

Signed-off-by: Thilo Fromm <thilo@kinvolk.io>

margamanterola reviewed (approved):
LGTM. Alright, let's see how this works.

t-lo commented:
Merged and [cherry-picked for v246-flatcar](https://github.com/flatcar-linux/systemd/commit/912af480e30f098a3c223a7249898efb688637ae).
